novaboot

```diff
@@ -1150,9 +1173,23 @@ host server {
 
 if (defined $dhcp_tftp || defined $tftp) {
     $tftp_port ||= 69;
+    my $tftp_root = "$builddir";
+    if (defined $server) {
+        $tftp_root = $tftp_root . "/$server";
+    }
+
+    # Prepare a GRUB netboot directory
+    system_verbose("grub-mknetdir --net-directory=$tftp_root");
+
+    # Generate TFTP mapfile
+    open(my $fh, '>', "$tftp_root/mapfile");
+    print $fh "# Some PXE clients (mainly UEFI) have bug. They add zero byte to the end of the
+# path name. This rule removes it
+r     \\.efi.*   \\.efi";
+    close($fh);
     # Unfortunately, tftpd requires root privileges even with
     # non-privileged (>1023) port due to initgroups().
-    system_verbose("sudo in.tftpd --listen --secure -v -v -v --pidfile tftpd.pid  --address :$tftp_port $builddir");
+    system_verbose("sudo in.tftpd --listen --secure -v -v -v --pidfile $builddir/tftpd.pid -m mapfile --address :$tftp_port $tftp_root");
```

Comment:
You should also update `sudoers.novaboot` file. I'm not sure why is `$builddir` added here.

Reply:
The `$builddir` is there because otherwise `in.tftpd` will create `tftpd.pid` in `$tftp_root` and then `pkill` will also need path to `$tftp_root/tftpd.pid`. I decided to leave it in the `$builddir` and not change path in the `pkill` because it's not part of the image/files used for loading and also `dhcpd.pid` is in `$builddir`. But I could change it to `$tftp_root`, it's not that important I think 😀

Reply:
My concern here is security. If I'm allowing to run something with `sudo` and don't want this ability to be misused. So generally, I want as many command arguments as possible to be specified in `sudoers` file so that the user cannot use arbitrary values there. If you remove `--pidfile tftpd.pid` from sudoers, than the user can run, e.g., `sudo in.tftpd ... --pidfile /etc/passwd` and overwrite system accounts database with garbage.

In this case, we can probably have `--pidfile */tftpd.pid` in sudoers and it will be sufficient.